01-reading.Rmd

```diff
@@ -614,23 +711,148 @@ Next, we tell R what page we want to scrape by providing the webpage's URL in qu
 page <- read_html("https://en.wikipedia.org/wiki/Canada")
 ```
 
+The `read_html` function directly downloads the source code for the page at 
+the URL you specify, just like your browser would if you navigated to that site. But 
+instead of  displaying the website to you, the `read_html` function just returns 
+the HTML source code itself, which we have
+stored in the `page` variable. Next, we send the page object to the `html_nodes`
+function, along with the CSS selectors we obtained from
+the SelectorGadget tool. Make sure to surround the selectors with quotations; `html_nodes` expects that
+argument to be a string. The `html_nodes` function then selects *nodes* from the HTML document that 
+match the CSS selectors you specified.
+A *node* is an HTML tag pair combined with the content stored between the tags. For our
+CSS selector `td:nth-child(5)`, an example node that would be selected would
+be: 
+```html
+<td style="text-align:left;background:#f0f0f0;"><a href="/wiki/London,_Ontario" title="London, Ontario">London</a></td>
+```
 
-Then we send the page object to the `html_nodes` function. We also provide that function with the CSS selectors we obtained from the selectorgadget tool. These should be surrounded by quotations. The `html_nodes` function select nodes from the HTML document using CSS selectors. Nodes are the HTML tag pairs as well as the content between the tags. For our CSS selector `td:nth-child(5)` and example node that would be selected would be: `<td style="text-align:left;background:#f0f0f0;"><a href="/wiki/London,_Ontario" title="London, Ontario">London</a></td>`
-
-We will use `head()` here to limit the print output of these vectors to 6 lines.
+We store the result of the `html_nodes` function in the `population_nodes` variable.
 ```{r 01-select-nodes}
 population_nodes <- html_nodes(page, "td:nth-child(5) , td:nth-child(7) , .infobox:nth-child(122) td:nth-child(1) , .infobox td:nth-child(3)")
 head(population_nodes)
```

Comment:
The original `population_nodes` line was too long (again worried for print), so the only way to shorten it AFAIK so it will not be cut off in print is to use `paste0`. I don't love this, but I don't know what else to do... Other options for breaking the code across lines adds a newline character to it. If we do this, we will need some sort of note explaining why `paste0` is there... Sigh... 

```suggestion
css_selector <- paste0("td:nth-child(5) , td:nth-child(7) , ",
                       ".infobox:nth-child(122) td:nth-child(1) , ",
                       ".infobox td:nth-child(3)")
population_nodes <- html_nodes(page, css_selector)
head(population_nodes)
```

Reply:
fixed with explanation